RealEstate/apps/core/views.py

```diff
@@ -219,3 +222,138 @@ def _permission_check(self, request, role, *args, **kwargs):
 
     def get(self, request, *args, **kwargs):
         return render(request, self.template_name, {})
+
+
+class CategoryView(BaseView):
+    """
+    View for the Category Ranking Page.
+    """
+    template_name = 'core/categories.html'
+
+    def _permission_check(self, request, role, *args, **kwargs):
+        return True
+
+    def _weight_context(self):
+        weight_field = CategoryWeight._meta.get_field('weight')
+        weight_choices = dict(weight_field.choices)
+        min_weight = min(weight for weight in weight_choices)
+        max_weight = max(weight for weight in weight_choices)
+        min_choice = weight_choices[min_weight]
+        max_choice = weight_choices[max_weight]
+        return {
+            'min_weight': min_weight,
+            'max_weight': max_weight,
+            'min_choice': min_choice,
+            'max_choice': max_choice,
+            'default_weight': weight_field.default,
+            'js_weight': json.dumps(weight_choices),
+        }
+
+    def get(self, request, *args, **kwargs):
+        
+        # Returns summary and description if given category ID
+        if request.is_ajax():
+            id = request.GET['category']
+            category = Category.objects.get(id=id)
+            response_data = {
+                'summary': category.summary,
+                'description': category.description
+            }
+            return HttpResponse(json.dumps(response_data),
+                                content_type="application/json")
+
+        # Renders standard category page
+        homebuyer = request.user.role_object
+        couple = Couple.objects.filter(homebuyer__user=request.user)
+        categories = Category.objects.filter(couple=couple)
+        weights = CategoryWeight.objects.filter(homebuyer__user=request.user)
```

Comment:
Should be able to filter on homebuyer instead of traversing the relationship all the way to user.
```python
weights = CategoryWeight.objects.filter(homebuyer=homebuyer)
```